crates/cli/cli-components/src/impls/commands/connection/create.rs

```diff
@@ -0,0 +1,145 @@
+use core::fmt::Display;
+use core::marker::PhantomData;
+
+use cgp::core::field::Index;
+use cgp::prelude::*;
+use hermes_logging_components::traits::has_logger::HasLogger;
+use hermes_logging_components::traits::logger::CanLog;
+use hermes_logging_components::types::level::LevelInfo;
+use hermes_relayer_components::build::traits::builders::relay_builder::CanBuildRelay;
+use hermes_relayer_components::chain::traits::types::chain_id::HasChainIdType;
+use hermes_relayer_components::chain::traits::types::connection::HasInitConnectionOptionsType;
+use hermes_relayer_components::chain::traits::types::ibc::HasClientIdType;
+use hermes_relayer_components::multi::traits::chain_at::HasChainTypeAt;
+use hermes_relayer_components::multi::traits::relay_at::HasRelayTypeAt;
+use hermes_relayer_components::relay::impls::connection::bootstrap::CanBootstrapConnection;
+use hermes_relayer_components::relay::traits::chains::HasRelayChains;
+
+use crate::traits::build::CanLoadBuilder;
+use crate::traits::command::CommandRunner;
+use crate::traits::output::CanProduceOutput;
+use crate::traits::parse::CanParseArg;
+
+pub struct RunCreateConnectionCommand;
+
+#[derive(Debug, clap::Parser, HasField)]
+pub struct CreateConnectionArgs {
+    #[clap(
+        long = "target-chain-id",
+        required = true,
+        value_name = "TARGET_CHAIN_ID",
+        help_heading = "REQUIRED"
+    )]
+    target_chain_id: String,
+
+    #[clap(
+        long = "target-client-id",
+        required = true,
+        value_name = "TARGET_CLIENT_ID",
+        help_heading = "REQUIRED"
+    )]
+    target_client_id: String,
+
+    #[clap(
+        long = "counterparty-chain-id",
+        required = true,
+        value_name = "COUNTERPARTY_CHAIN_ID",
+        help_heading = "REQUIRED"
+    )]
+    counterparty_chain_id: String,
+
+    #[clap(
+        long = "counterparty-client-id",
+        required = true,
+        value_name = "COUNTERPARTY_CLIENT_ID",
+        help_heading = "REQUIRED"
+    )]
+    counterparty_client_id: String,
+}
+
+impl<App, Args, Builder, Chain, Counterparty, Relay> CommandRunner<App, Args>
+    for RunCreateConnectionCommand
+where
+    App: CanLoadBuilder<Builder = Builder>
+        + HasLogger
+        + CanProduceOutput<&'static str>
+        + CanRaiseError<Builder::Error>
+        + CanRaiseError<Relay::Error>
+        + CanParseArg<Args, symbol!("target_chain_id"), Parsed = Chain::ChainId>
+        + CanParseArg<Args, symbol!("counterparty_chain_id"), Parsed = Counterparty::ChainId>
+        + CanParseArg<Args, symbol!("target_client_id"), Parsed = Chain::ClientId>
+        + CanParseArg<Args, symbol!("counterparty_client_id"), Parsed = Counterparty::ClientId>,
+    App::Logger: CanLog<LevelInfo>,
+    Builder: CanBuildRelay<Index<0>, Index<1>, Relay = Relay>
```

Comment:
Sorry in case you got confusing error here: you imported the `Index` type from `cgp-field`, while we still have our own copy of `Index` type due to the upstream `Index` didn't implement `Display`.

I will upgrade the `cgp` crate tomorrow, so that we can use the upstream `Index` type and avoid confusing errors like this.